modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache

```diff
@@ -19,15 +19,23 @@ class URLSessionRequestBuilderFactory: RequestBuilderFactory {
     }
 }
 
+public typealias {{projectName}}APIChallengeHandler = ((URLSession, URLSessionTask, URLAuthenticationChallenge) -> (URLSession.AuthChallengeDisposition, URLCredential?))
+
 // Store the URLSession to retain its reference
-private var urlSessionStore = SynchronizedDictionary<String, URLSession>()
+private var urlSessionStore = SynchronizedDictionary<Int, URLSession>()
+
+// Store the URLSession's delegate to retain its reference
+private var sessionDelegateStore = SynchronizedDictionary<Int, SessionDelegate>()
+
+// Store current taskDidReceiveChallenge for every URLSessionTask
+private var challengeHandlerStore = SynchronizedDictionary<Int, {{projectName}}APIChallengeHandler>()
```

Comment:
@DavidHorvath since the SessionDelegate is now shared, we also need to store the credential associated with the task identifier
```
// Store current URLCredential for every URLSessionTask
 private var credentialStore = SynchronizedDictionary<Int, URLCredential >()
```